### ISSUE TITLE
Migration from GCM to FCM: inconsistent?

### ISSUE THREAD

**Issue by rseibane** [QuickBlox/quickblox-android-sdk#390]
Hello,

I just wanna explain the reasons why I feel you need to improve the support for FCM. The following points are based on the case you are using FCM on your project.

- Clarify the setup guide found at https://quickblox.com/developers/SimpleSample-messages_users-android#Setup_GCM. If I understand, after step 3 the device is ready to receive notifications, isn't it? In the real scenario I realize that step 6 is mandatory. Concretely, the part of subscribe to the service. if you don't register a `FirebaseInstanceIdService` and call `SubscribeService.subscribeToPushes(context, true);` you won't be ready to receive push notifications. Please, clarify this if I'm wrong.

- The part that pisses me. After that, when I receive the invocation to my receiver on `public void onReceive(Context context, Intent intent)` I can access to two fields: 'message' and 'from', whereas 'from' corresponds to the Firebase SenderID. Isn't it? So, what the hell should I do with the SenderID? On the GCM alternative I could receive directly the QuickBlox UserID. Why not now? Please, explain me!

- Other inconvenient I found: how can I receive custom fields with the notification? As far as I concern, `public void onReceive(Context context, Intent intent)` receives always two fields: 'from' and 'message'. If it's impossible to receive custom parameters, there is a inconsistency between send custom fields with a QBEvent and receive them. What's about this?

- Not mandatory, of course, but It'd be great if you'd provide a project sample about FCM.

Thanks for be there.
Regards,

**Comment by RomanPronin** [comment]
Hello @rafa0809, thank you for your comment. 
At first, after step 3 the device is really ready to receive notifications, and there is a notion about QBGcmPushInstanceIDService in first [step](https://quickblox.com/developers/SimpleSample-messages_users-android#Steps_to_setup_SDK_Settings.), naturally this also applies to QBFcmPushInstanceIDService.
Secondly, you can extend QBFcmPushListenerService and  override sendPushMessage(Map data, String from, String message), where data - is data with your custom fields.
And we will think about providing sample with FCM as soon as we can.

**Comment by rafa0809** [comment]
Hello @RomanPronin !
Sorry If I'm wrong, I'm not native english speaker and I could be wrong but as far as I concern, it means for me that I just need to include one piece of code and not to guess any more. 
Documentation words: "Note, if you are using type FCM, it is a little bit easier, to get message **you just need** set FcmPushListenerService:"

` <service android:name="com.quickblox.messages.services.fcm.QBFcmPushListenerService">
            <intent-filter>
                 <action android:name="com.google.firebase.MESSAGING_EVENT" />
            </intent-filter>
    </service>`

About step 2, I'm aware that I can extend QBFcmPushListenerService, of course. But the data provided by the intent that you sent to the broadcast receiver isn't very powerful (just the message from the chat but not from who, which dialog, etc....). The purpose of integrate an SDK is to avoid, as much as possible, replicating some obvious work and use them to go to the point.

I would like to know what's in your mind about this comments. Basically, if you plan to change the intent of the notification receiver. Please, let me know.

Thanks in advance,

**Comment by RomanPronin** [comment]
@rafa0809, we've updated documents. Also, we will think about changing the intent of the notification receiver. Thanks for suggestions.

**Comment by alexanderkoqb** [comment]
Hey @rseibane 
Since it's been a while and if you're still being experiencing troubles with the newer SDK (WebRTC) versions, you can contact QB here: https://assist.quickblox.com/index.php?/Base/UserRegistration/Register or raise a new issue in corresponding GitHub project.
Best!
